apps/src/lab2/projects/ProjectManager.ts

```diff
@@ -489,6 +495,44 @@ export default class ProjectManager {
     }
   }
 
+  /**
+   * Load the sources for this project. If a versionId is provided, load that version, otherwise
+   * load the latest version.
+   * @param versionId Optional version id to load. If not provided, the latest version is loaded.
+   * @param isPreview Optional boolean to indicate if we are previewing a project. If we are previewing,
```

Comment:
Thanks for the comments! This makes sense - just as an alternate idea, we could also maybe have two separate functions, one that just loads the project for a given version and another that calls load and saves it to `lastSources`? Something like

```
private async loadSources(versionId?: string): Promise<ProjectSources | undefined> {
   ... existing load code, without storing in this.lastSource
}

private async loadAndStoreSources(versionId?: string) {
   const sources = await loadSources(versionId);
   this.lastSource = JSON.stringify(sources);
   return sources;
}
```

Then `load` and `restore` can call `loadAndStoreSources` while any previewing APIs can just call `loadSources()`? Might be overkill, but just as a thought to reduce ambiguity. This works too though, so not blocking

Reply:
Good idea! I might add that in the follow-up to do view-only, since that will be the use case for just loading sources.